source/common/upstream/cluster_manager_impl.cc

```diff
@@ -301,7 +301,7 @@ ClusterManagerImpl::ClusterManagerImpl(const envoy::config::bootstrap::v2::Boots
   if (cm_config.has_load_stats_config()) {
     const auto& load_stats_config = cm_config.load_stats_config();
     load_stats_reporter_.reset(
-        new LoadStatsReporter(bootstrap.node(), *this, stats,
+        new LoadStatsReporter(local_info.node(), *this, stats,
```

Comment:
Another option, not saying you have to do this, is to treat the CLI flags as a bootstrap override and override the bootstrap proto early in server config, so that `local_info` and `bootstrap` are consistent. Or, maybe nuke node from bootstrap once you've loaded up local_info. Ideally we have a single source-of-truth, and it's hard to confuse with other potential sources.